### PR TITLE
[CloudKit] Update bindings for Xcode 8.2

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/cloudkit.cs [hunk 1]
```diff
@@ -960,11 +959,7 @@ interface CKNotification : NSSecureCoding {
 
 		[Static]
 		[Export ("notificationFromRemoteNotificationDictionary:")]
-#if XAMCORE_4_0
```
Comment (dalexsoto) [inline src/cloudkit.cs]: Just curious, why did you remove this code?

Comment (mandel-macaque) [inline src/cloudkit.cs]: Cause I added it due to changes in the API that typed the dictionary but Apple reverted that change. Ergo, I remove it to make sure that we do not change the API.
